addons/mail/models/mail_thread_main_attachment.py

```diff
@@ -21,11 +21,21 @@ def _message_post_after_hook(self, message, msg_values):
             for attachment_command in (msg_values['attachment_ids'] or [])
         ])
 
-    def _message_set_main_attachment_id(self, attachment_ids):
-        if attachment_ids and not self.message_main_attachment_id:
+    def _message_set_main_attachment_id(self, attachment_ids, force=False, filter_xml=True):
```

Comment:
Looks like all calls (except one I think?) have attachments and purposefully call ".ids", to be re-browsed here.

Was wondering if we could not change to "_message_set_main_attachment" and make it take a browserecord instead.

Reply:
Private functions should definitely take recordsets 👍
But is there a performance issue in this case?